src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -656,6 +656,11 @@ private BoundExpression BindExpressionInternal(ExpressionSyntax node, Diagnostic
             }
         }
 
+        private static BoundExpression BindDefaultLiteral(DefaultLiteralSyntax node)
+        {
+            return new BoundDefaultLiteral(node, ConstantValue.DefaultLiteral, type: null);
+        }
```

Comment:
I will check. I have a test for `int x = (short)default;` (legal), but not with a generic cast. 
My expectation is that `(T)default` should behave the same as `default(T)`.